lib/state.js

```diff
@@ -349,7 +364,7 @@ StateResolver.prototype.extractArgumentsList_ = function (frame) {
 StateResolver.prototype.resolveArgumentList_ = function(args) {
   var resolveVariable = this.resolveVariable_.bind(this);
```

Comment:
Until [recently](http://benediktmeurer.de/2016/01/14/optimizing-bound-functions-further/) bind used to be quite slow. With Node v7.x there should be no difference in performance but earlier versions should have different degrees of performance issues with `bind`.

Having said this, there are other occurrences of `bind` in this file, and I'll let it up to change them in a follow-on change.